vm/host/host_test.go

```diff
@@ -0,0 +1,130 @@
+package host
+
+import (
+	"encoding/binary"
+	"os"
+	"testing"
+
+	athcon "github.com/athenavm/athena/ffi/athcon/bindings/go"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql/statesql"
+	"github.com/spacemeshos/go-spacemesh/vm/core"
+	"github.com/spacemeshos/go-spacemesh/vm/programs/getbalance"
+	hostprogram "github.com/spacemeshos/go-spacemesh/vm/programs/host"
+)
+
+func getHost(t *testing.T) (*Host, *core.StagedCache) {
+	os.Setenv("ATHENA_LIB_PATH", "../../build")
+	cache := core.NewStagedCache(core.DBLoader{Executor: statesql.InMemoryTest(t)})
+	logger, err := zap.NewDevelopment()
+	require.NoError(t, err)
+	ctx := &core.Context{Loader: cache, Logger: logger}
+	host, err := NewHost(ctx)
+	require.NoError(t, err)
+	return host, cache
+}
+
+func TestNewHost(t *testing.T) {
+	host, _ := getHost(t)
+	defer host.Destroy()
+
+	require.Equal(t, "Athena", host.vm.Name())
+}
+
+func TestGetBalance(t *testing.T) {
+	host, cache := getHost(t)
+	defer host.Destroy()
+
+	account := types.Account{
+		Layer:   types.LayerID(15),
+		Address: types.Address{1, 2, 3, 4},
+		Balance: 100,
+	}
+	err := cache.Update(account)
+	require.NoError(t, err)
+
+	out, gasLeft, err := host.Execute(
+		account.Layer,
+		10000,
+		account.Address,
+		account.Address,
+		nil,
+		0,
+		getbalance.PROGRAM,
+	)
+
+	require.NoError(t, err)
+	balance := binary.LittleEndian.Uint64(out)
+	require.Equal(t, account.Balance, balance)
+	require.NotZero(t, gasLeft)
+}
+
+func TestNotEnoughGas(t *testing.T) {
+	host, _ := getHost(t)
+	defer host.Destroy()
+
+	_, gasLeft, err := host.Execute(
+		10,
+		10,
+		types.Address{1, 2, 3, 4},
+		types.Address{1, 2, 3, 4},
+		nil,
+		0,
+		getbalance.PROGRAM,
+	)
+
+	require.ErrorIs(t, err, athcon.OutOfGas)
+	require.Zero(t, gasLeft)
+}
+
+func TestEmptyCode(t *testing.T) {
+	host, _ := getHost(t)
+	defer host.Destroy()
+
+	_, _, err := host.Execute(
+		10,
+		10,
+		types.Address{1, 2, 3, 4},
+		types.Address{1, 2, 3, 4},
+		nil,
+		0,
+		[]byte{},
+	)
+
+	require.ErrorContains(t, err, "athcon execute: no input code")
+}
```

Comment:
It's not testing any functionality implemented in `go-spacemesh` but in the Athena bindings. I'd remove it.

Reply:
but that's testing the contract between node and athena lib. imo it's fine.

Reply:
can we say it tests whether errors are correctly surfaced from VM -> host? :)